parser/src/lexer.rs

```diff
@@ -494,6 +497,55 @@ where
         Ok(())
     }
 
+    /// Lex a single magic command.
+    fn lex_magic_command(&mut self, kind: MagicKind) -> LexResult {
+        let start_pos = self.get_pos();
+        for _ in 0..u32::from(kind.prefix_len()) {
+            self.next_char();
+        }
+        let mut value = String::new();
+        loop {
+            match self.window[0] {
+                Some('\\') => {
+                    // Only skip the line continuation if it is followed by a newline
+                    // otherwise it is a normal backslash which is part of the magic command:
+                    //
+                    //        Skip this backslash
+                    //        v
+                    //   !pwd \
+                    //      && ls -a | sed 's/^/\\    /'
+                    //                          ^^
+                    //                          Don't skip these backslashes
+                    if matches!(self.window[1], Some('\n' | '\r')) {
```

Comment:
Should `value` include the `\` and `\n` sequence? Does ipython trim any indent after a continuation

Reply:
No, the `value` doesn't include `\` and `\n` sequence nor does IPython trim indentation. Consider the following code:

```python
from IPython.core.inputtransformer2 import TransformerManager

s = r"""%foo \
    arg"""

t = TransformerManager()
t.transform_cell(s)
# "get_ipython().run_line_magic('foo', '     arg')\n"
```

_The code can be run as is in the IPython REPL_

Reply:
Oh. It isn't necessary because `value.push(self.next_char().unwrap());` calls `self.next_char()`). Should we add a `continue` after line 521 to make it explicit that we skip the characters. 

Could the current code panic if you have a magic with a continuation token at the end of the document 

```python
%test \\n
EOF
```

If so, then we'll need to explicitly handle `\r\n` here too